infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -134,13 +135,21 @@ public void onEntry(LogReplicationState from) {
         this.fsm.getLogReplicationFSMWorkers().submit(this::verifyStatusOfSnapshotSyncApply);
```

Comment:
If I understand correctly, as per the FSM, it is possible to enter this state from the INITIALIZED state.  If so, is it valuable to update the replication status to 'snapshot sync'?

Reply:
Good catch, let me see how I make this cleaner, instead of update all over the place.